src/sqlfluff/dialects/dialect_hive.py

```diff
@@ -687,7 +687,6 @@ class FunctionSegment(BaseSegment):
                 Delimited(
                     Sequence(
                         Ref("BaseExpressionElementGrammar"),
-                        Ref("DatatypeIdentifierSegment", optional=True),
```

Comment:
this one not needed cause `row` as a function here accepting only sequence of expressions